src/geometry/rtree.cpp

```diff
@@ -16,12 +16,14 @@ namespace GeoShape {
 
 namespace {
 template <typename cs, typename rect_type = RTree<cs>::rect_type>
-constexpr auto make_mbr = [](auto&& geom) -> rect_type {
+constexpr auto make_mbr = [](auto const& geom) -> rect_type {
   using point_type = typename RTree<cs>::point_type;
   if constexpr (std::is_same_v<point_type, std::decay_t<decltype(geom)>>) {
-    constexpr auto EPSILON = 1e-10;
-    auto p1 = point_type{bg::get<0>(geom) - EPSILON, bg::get<1>(geom) - EPSILON};
-    auto p2 = point_type{bg::get<0>(geom) + EPSILON, bg::get<1>(geom) + EPSILON};
+    constexpr auto INF = std::numeric_limits<long double>::infinity();
+    const auto x = bg::get<0>(geom);
+    const auto y = bg::get<1>(geom);
+    const auto p1 = point_type{std::nexttoward(x, -INF), std::nexttoward(y, -INF)};
+    const auto p2 = point_type{std::nexttoward(x, +INF), std::nexttoward(y, +INF)};
```

Comment:
Consider an enhancement to improve accuracy to use 4 docs per point each with one quarter of the (p1,p2) rect